app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/OwnedViewModel.java

```diff
@@ -1,101 +1,104 @@
 package com.cmput301f20t21.bookfriends.ui.library;
 
-import android.net.Uri;
 
 import androidx.lifecycle.LiveData;
 import androidx.lifecycle.MutableLiveData;
 import androidx.lifecycle.ViewModel;
 
-import com.cmput301f20t21.bookfriends.callbacks.OnFailCallback;
-import com.cmput301f20t21.bookfriends.callbacks.OnSuccessCallbackWithMessage;
 import com.cmput301f20t21.bookfriends.entities.Book;
 import com.cmput301f20t21.bookfriends.enums.BOOK_ERROR;
 import com.cmput301f20t21.bookfriends.repositories.AuthRepository;
 import com.cmput301f20t21.bookfriends.repositories.BookRepository;
 import com.google.firebase.firestore.DocumentSnapshot;
-import com.google.firebase.firestore.QuerySnapshot;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
 public class OwnedViewModel extends ViewModel {
-  
+
     private final AuthRepository authRepository = AuthRepository.getInstance();
     private final BookRepository bookRepository = BookRepository.getInstance();
-    
-    private MutableLiveData<List<Book>> books;
-    private MutableLiveData<Integer> updatedPosition;
-    private MutableLiveData<BOOK_ERROR> errorMessageObserver;
+
+    private MutableLiveData<List<Book>> books = new MutableLiveData<>(new ArrayList<>());
+    private List<Book> bookData = books.getValue();
```

Comment:
should we make this local var? we saved a bunch of getValue call but now it's easier to lost sync with bookData and books livedata.

Reply:
I'm sure this is by reference as well so books.getValue() will always == bookData. We won't be out of sync but I think observe only gets called when we call books.setValue()